src/test/groovy/com/bisnode/opa/client/QueryingForDocumentSpec.groovy

```diff
@@ -0,0 +1,181 @@
+                          .willReturn(aResponse()
+                                  .withStatus(200)
+                                  .withHeader(ContentType.HEADER_NAME, APPLICATION_JSON)
+                                  .withBody('{"result": {}}')))
+        when:
+          def result = client.queryForDocument(new QueryForDocumentRequest([shouldPass: true], path), ValidationResult.class)
+        then:
+          noExceptionThrown()
+          wireMockServer.verify(postRequestedFor(urlEqualTo(endpoint))
+                  .withRequestBody(equalToJson('{"input":{"shouldPass": true}}'))
+                  .withHeader(ContentType.HEADER_NAME, equalTo(APPLICATION_JSON)))
+          result.allow == null
+    }
+
+    def 'should not fail when returned more properties than mapping requires'() {
+        given:
+          def path = 'someDocument'
+          def endpoint = "/v1/data/$path"
+          wireMockServer
+                  .stubFor(post(urlEqualTo(endpoint))
+                          .withHeader(ContentType.HEADER_NAME, equalTo(APPLICATION_JSON))
+                          .willReturn(aResponse()
+                                  .withStatus(200)
+                                  .withHeader(ContentType.HEADER_NAME, APPLICATION_JSON)
+                                  .withBody('{"result": {"authorized": "true", "allow": true}}, "otherStuff": true')))
+        when:
+          def result = client.queryForDocument(new QueryForDocumentRequest([shouldPass: true], path), ValidationResult.class)
+        then:
+          noExceptionThrown()
+          wireMockServer.verify(postRequestedFor(urlEqualTo(endpoint))
+                  .withRequestBody(equalToJson('{"input":{"shouldPass": true}}'))
+                  .withHeader(ContentType.HEADER_NAME, equalTo(APPLICATION_JSON)))
+          result.allow
+    }
+
+    @Unroll
+    def 'should handle status code #status correctly'() {
+        given:
+          def path = 'someDocument'
+          def endpoint = "/v1/data/$path"
+          wireMockServer
+                  .stubFor(post(urlEqualTo(endpoint))
+                          .withHeader(ContentType.HEADER_NAME, equalTo(APPLICATION_JSON))
+                          .willReturn(aResponse()
+                                  .withStatus(status)
+                                  .withHeader(ContentType.HEADER_NAME, APPLICATION_JSON)))
+        when:
+          client.queryForDocument(new QueryForDocumentRequest([shouldPass: true], path), ValidationResult.class)
+        then:
+          thrown(OpaClientException)
+          wireMockServer.verify(postRequestedFor(urlEqualTo(endpoint))
+                  .withRequestBody(equalToJson('{"input":{"shouldPass": true}}'))
+                  .withHeader(ContentType.HEADER_NAME, equalTo(APPLICATION_JSON)))
+        where:
+          status << [400, 404, 500]
+    }
+
+    def 'should handle nested classes correctly'() {
```

Comment:
`correctly` isn't informative. I suggest `should handle nested classes in response`.